bpf/process/bpf_execve_event.c

```diff
@@ -126,6 +127,71 @@ read_args(void *ctx, struct msg_execve_event *event)
 	return size;
 }
 
+#ifdef __LARGE_BPF_PROG
+FUNC_INLINE __u32 read_envs(void *ctx, struct msg_execve_event *event)
+{
+	struct msg_process *p = &event->process;
+	struct tetragon_conf *conf;
+	struct mm_struct *mm = NULL;
+	struct task_struct *task;
+	__u32 size = 0, flags = 0, key = 0;
+	unsigned long free_size, envs_size;
+	unsigned long env_start, env_end;
+	char *envs;
+	int err;
+
+	conf = map_lookup_elem(&tg_conf_map, &key);
+	if (!conf || !conf->env_vars_enabled)
+		return 0;
+
+	task = (struct task_struct *)get_current_task();
+	probe_read(&mm, sizeof(mm), _(&task->mm));
+	if (!mm)
+		return 0;
+
+	with_errmetrics(probe_read, &env_start, sizeof(env_start), _(&mm->env_start));
+	with_errmetrics(probe_read, &env_end, sizeof(env_end), _(&mm->env_end));
+
+	if (!env_start || !env_end)
+		return 0;
+
+	envs = (char *)p + p->size;
+	if (envs >= (char *)&event->process + BUFFER)
+		return 0;
```

Comment:
I'd move this before the `probe_read` to avoid wasting time on reading env start and end when not needed.